invenio_users_resources/records/api.py

```diff
@@ -214,13 +216,29 @@ def avatar_color(self):
     @classmethod
     def create(cls, data, id_=None, validator=None, format_checker=None, **kwargs):
         """Create a new User and store it in the database."""
-        # NOTE: we don't use an actual database table, and as such can't
-        #       use db.session.add(record.model)
-        with db.session.begin_nested():
-            # create_user() will already take care of creating the profile
-            # for us, if it's specified in the data
-            user = current_datastore.create_user(**data)
-            return cls.from_model(user)
+        try:
+            # Check if email and  username already exists by another account.
+            errors = {}
+            existing_email = (
+                db.session.query(User).filter_by(email=data["email"]).first()
+            )
+            if existing_email:
+                errors["email"] = ["Email already used by another account."]
+            existing_username = (
+                db.session.query(User).filter_by(username=data["username"]).first()
+            )
+            if existing_username:
+                errors["username"] = ["Username already used by another account."]
+            if errors:
+                raise ValidationError(errors)
+            # Create User
+            account_user = current_datastore.create_user(**data)
+            current_datastore.commit()  # Commit to save the user to the database
```

Comment:
Question: Should we Consider adding a rollback or transaction management for unexpected failures?

Reply:
🤔  Wouldn't the transaction roll back once the exception is raised? i.e., line 241

Reply:
i think only if the function is wrapped by the uow otherwise not. assuming you are referring to https://github.com/inveniosoftware/invenio-db/blob/305b5b179c1b9441f770c64bba2332436a79593e/invenio_db/uow.py#L171

Reply:
This was on an oversite! We did have `RecordCommitOp` at some point and removed it to check what errors we would get after saving, i.e. duplicate emails/username. Then we added the control code for that and forgot to put back the `RecordCommitOp` ... 😇